server/src/main/java/keywhiz/service/filters/SecurityHeadersFilter.java

```diff
@@ -36,24 +39,35 @@
  *   IE X-Content-Type-Options: http://msdn.microsoft.com/en-us/library/ie/gg622941(v=vs.85).aspx
  *   IE X-XSS-Protection: http://blogs.msdn.com/b/ie/archive/2008/07/02/ie8-security-part-iv-the-xss-filter.aspx
  */
-public class SecurityHeadersFilter implements ContainerResponseFilter {
-  private static final long YEAR_OF_SECONDS = TimeUnit.DAYS.convert(365, TimeUnit.SECONDS);
+public class SecurityHeadersFilter implements Filter {
+  private static final long YEAR_OF_SECONDS = TimeUnit.SECONDS.convert(365, TimeUnit.DAYS);
 
-  @Override public void filter(ContainerRequestContext request, ContainerResponseContext response)
-      throws IOException {
-    MultivaluedMap<String, Object> headers = response.getHeaders();
+  @Override public void init(FilterConfig filterConfig) throws ServletException {
+  }
+
+  @Override public void doFilter(ServletRequest request, ServletResponse response,
+      FilterChain chain) throws IOException, ServletException {
+    if (response instanceof HttpServletResponse) {
+      HttpServletResponse r = (HttpServletResponse) response;
+
+      // Defense against XSS. We don't care about IE's Content-Security-Policy because it's useless
+      r.addHeader("X-Content-Security-Policy", "default-src 'self'");
+      r.addHeader(HttpHeaders.X_XSS_PROTECTION, "0"); // With CSP, we don't need crazy magic
 
-    // The X-* CSP headers are transitional and used by some versions of Firefox and Chrome.
-    headers.add(HttpHeaders.CONTENT_SECURITY_POLICY, "default-src 'self'");
-    headers.add("X-Content-Security-Policy", "default-src 'self'");
-    headers.add("X-WebKit-CSP", "default-src 'self'");
+      // Tell IE not to do silly things
+      r.addHeader(HttpHeaders.X_CONTENT_TYPE_OPTIONS, "nosniff");
 
-    headers.add("Frame-Options", "DENY");
-    headers.add(HttpHeaders.X_FRAME_OPTIONS, "DENY");
+      // Protection against click jacking
+      r.addHeader("Frame-Options", "DENY"); // Who uses this?
+      r.addHeader(HttpHeaders.X_FRAME_OPTIONS, "DENY");
+
+      // https-all-the-time
+      r.addHeader(HttpHeaders.STRICT_TRANSPORT_SECURITY, format("max-age=%d; includeSubDomains", YEAR_OF_SECONDS));
+
+      chain.doFilter(request, response);
```

Comment:
Fixed. https://dropwizard.github.io/dropwizard/manual/core.html#servlet-filters does the chain.doFilter inside the if statement.